packages/yswaps/contracts/TradeFactory/TradeFactoryPositionsHandler.sol

```diff
@@ -60,73 +55,67 @@ abstract contract TradeFactoryPositionsHandler is ITradeFactoryPositionsHandler,
     _setupRole(STRATEGY_ADDER, _strategyAdder);
   }
 
-  function tradeDetails() external view override returns (TradeDetail[] memory _tradeDetailsList) {
-    uint256 _tradeAmount;
+  function enabledTrades() external view override returns (TradeDetail[] memory _enabledTrades) {
+    uint256 _totalEnabledTrades;
     for (uint256 i; i < _strategies.values().length; i++) {
       address _strategy = _strategies.at(i);
       address[] memory _tokensIn = _tokensInByStrategy[_strategy].values();
       for (uint256 j; j < _tokensIn.length; j++) {
         address _tokenIn = _tokensIn[j];
-        address[] memory _tokensOut = _tokenOutsByStrategyAndTokenIn[_strategy][_tokenIn].values();
-        _tradeAmount += _tokensOut.length;
+        _totalEnabledTrades += _tokensOutByStrategyAndTokenIn[_strategy][_tokenIn].length();
       }
     }
-    _tradeDetailsList = new TradeDetail[](_tradeAmount);
-    uint256 _tradeDetailsIndex;
+    _enabledTrades = new TradeDetail[](_totalEnabledTrades);
+    uint256 _enabledTradesIndex;
     for (uint256 i; i < _strategies.values().length; i++) {
       address _strategy = _strategies.at(i);
       address[] memory _tokensIn = _tokensInByStrategy[_strategy].values();
       for (uint256 j; j < _tokensIn.length; j++) {
         address _tokenIn = _tokensIn[j];
-        address[] memory _tokensOut = _tokenOutsByStrategyAndTokenIn[_strategy][_tokenIn].values();
+        address[] memory _tokensOut = _tokensOutByStrategyAndTokenIn[_strategy][_tokenIn].values();
         for (uint256 k; k < _tokensOut.length; k++) {
-          address _tokenOut = _tokensOut[k];
-          _tradeDetailsList[_tradeDetailsIndex] = TradeDetail(_strategy, _tokenIn, _tokenOut);
-          _tradeDetailsIndex++;
+          _enabledTrades[_enabledTradesIndex] = TradeDetail(_strategy, _tokenIn, _tokensOut[k]);
+          _enabledTradesIndex++;
         }
       }
     }
   }
 
-  function create(address _tokenIn, address _tokenOut) external override onlyRole(STRATEGY) returns (bool _success) {
+  function enable(address _tokenIn, address _tokenOut) external override onlyRole(STRATEGY) {
     if (_tokenIn == address(0) || _tokenOut == address(0)) revert CommonErrors.ZeroAddress();
     _strategies.add(msg.sender);
     _tokensInByStrategy[msg.sender].add(_tokenIn);
-    if (!_tokenOutsByStrategyAndTokenIn[msg.sender][_tokenIn].add(_tokenOut)) revert InvalidTrade();
-    emit TradeCreated(msg.sender, _tokenIn, _tokenOut);
-    return true;
+    _tokensOutByStrategyAndTokenIn[msg.sender][_tokenIn].add(_tokenOut);
```

Comment:
Since we are not checking in all the other ones, I didn't see the need to revert here. If it already exists, then it will just emit the event (no biggie). If it didn't exist, it will add it.

Reply:
hmm i like having reverts on these functions.
the gas savings are minimal, and UX is more important on this function

Reply:
re-enabled revert sir. @skeletor-spaceman